google/cloud/testing_util/opentelemetry.h

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_TESTING_UTIL_OPENTELEMETRY_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_TESTING_UTIL_OPENTELEMETRY_H
+
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
+#include "google/cloud/version.h"
+#include <gmock/gmock.h>
+#include <opentelemetry/exporters/memory/in_memory_span_exporter.h>
+#include <opentelemetry/sdk/instrumentationscope/instrumentation_scope.h>
+#include <opentelemetry/sdk/trace/simple_processor.h>
+#include <opentelemetry/sdk/trace/span_data.h>
+#include <opentelemetry/sdk/trace/tracer.h>
+#include <opentelemetry/sdk/trace/tracer_provider_factory.h>
+#include <opentelemetry/trace/default_span.h>
+#include <opentelemetry/trace/provider.h>
+#include <opentelemetry/version.h>
+
+namespace google {
+namespace cloud {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace testing_util {
+
+using SpanDataPtr = std::unique_ptr<opentelemetry::sdk::trace::SpanData>;
+
+::testing::Matcher<SpanDataPtr> SpanHasInstrumentationScope();
+
+::testing::Matcher<SpanDataPtr> SpanKindIsClient();
+
+::testing::Matcher<SpanDataPtr> SpanNamed(std::string const& name);
+
+// A test fixture that provides access to created spans.
+class OpenTelemetryTest : public ::testing::Test {
+ protected:
+  OpenTelemetryTest();
+
+  std::shared_ptr<opentelemetry::exporter::memory::InMemorySpanData> span_data_;
```

Comment:
I think a standalone function (returning `std::shared_ptr<...::InMemorySpanData>`) is a cleaner start.

Creating a test fixture, just to initialize one member variable, yields less readable tests over time.

Reply:
Fixed.